util/journaldb/src/overlayrecentdb.rs

```diff
@@ -279,11 +280,19 @@ impl JournalDB for OverlayRecentDB {
 	fn earliest_era(&self) -> Option<u64> { self.journal_overlay.read().earliest_era }
 
 	fn state(&self, key: &H256) -> Option<Bytes> {
-		let journal_overlay = self.journal_overlay.read();
 		let key = to_short_key(key);
-		journal_overlay.backing_overlay.get(&key, EMPTY_PREFIX)
-			.or_else(|| journal_overlay.pending_overlay.get(&key).map(|d| d.clone()))
-			.or_else(|| self.backing.get_by_prefix(self.column, &key[0..DB_PREFIX_LEN]).map(|b| b.to_vec()))
+		let maybe_state_data = {
+			let journal_overlay = self.journal_overlay.try_read_for(Duration::from_secs(2))?;
```

Comment:
q: have you seen any errors for this? is this primarily used for private-transactions or am reading the code wrong?

Reply:
No, no proof, this is completely conjecture on my part. The idea is: if we need more than 2 seconds to get a lock, something's not right and we're better off bailing.

The `state_data()` is used primarily to respond to `GetNodeData` requests (only FastSync nodes emit such requests) and for light clients to fetch contract data. (And also from ipfs, but not sure we care so much about that.)

Reply:
I'm fine reverting to the previous `read()` here if that makes it easier to get this merged.

Reply:
yeah, I think reverting this is better than otherwise I would want to to use `try_read_for` for every read on `journal_overlay` because it is used by some other methods too.